src/server/worker/datum/iterator.go

```diff
@@ -77,12 +78,7 @@ func newPFSIterator(pachClient *client.APIClient, input *pps.PFSInput) (Iterator
 	// not possible for 2 inputs to have the same path so this is guaranteed to
 	// produce a deterministic order.
 	sort.Slice(result.inputs, func(i, j int) bool {
-		// We sort by descending size first because it can boost performance to
-		// process the biggest datums first.
-		if result.inputs[i].FileInfo.SizeBytes != result.inputs[j].FileInfo.SizeBytes {
-			return result.inputs[i].FileInfo.SizeBytes > result.inputs[j].FileInfo.SizeBytes
-		}
-		return result.inputs[i].FileInfo.File.Path < result.inputs[j].FileInfo.File.Path
+		return bytes.Compare(result.inputs[i].FileInfo.Hash, result.inputs[j].FileInfo.Hash) < 0
```

Comment:
This was one of the most hated features of the existing system because it intentionally grouped all of the largest datums into the same chunk which frequently led to stragglers. It wasn't strictly necessary to make the rest of autoscaling work, but it seemed like a good opportunity to fix it (honestly it was way overdue) and might help with people's overall perception of the autoscaling changes.